src/avlizer.erl

```diff
@@ -25,17 +25,73 @@
         ]).
 
 start(_StartType, _StartArgs) ->
-  case osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false) of
-    false -> ok;
-    URL ->
-      Vars0 = application:get_env(?APPLICATION, avlizer_confluent, #{}),
-      Vars = Vars0#{schema_registry_url => URL},
-      application:set_env(?APPLICATION, avlizer_confluent, Vars)
-  end,
+  set_envs(),
   avlizer_sup:start_link().
 
 stop(_State) -> ok.
 
+set_envs() ->
+  Vars0 = application:get_env(?APPLICATION, avlizer_confluent, #{}),
+  Vars1 = maybe_put_schema_registry_url(Vars0),
+  Vars2 = maybe_put_schema_registry_auth(Vars1),
+  application:set_env(?APPLICATION, avlizer_confluent, Vars2).
+
+maybe_put_schema_registry_url(Map) ->
+  SchemaRegistryURL = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_URL", false),
+  case SchemaRegistryURL of
+    false -> Map;
+    URL -> maps:put(schema_registry_url, URL, Map)
+  end.
+
+maybe_put_schema_registry_auth(Map) ->
+  SchemaRegistryAuthMechanism = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_MECHANISM", false),
+  SchemaRegistryAuthUsername = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_USERNAME", false),
+  SchemaRegistryAuthPassword = osenv("AVLIZER_CONFLUENT_SCHEMAREGISTRY_AUTH_PASSWORD", false),
+
+  Auth = {maybe_string_to_atom(SchemaRegistryAuthMechanism), SchemaRegistryAuthUsername, SchemaRegistryAuthPassword},
+
+  Map0 = case lists:member(false, tuple_to_list(Auth)) of
+    true -> Map;
+    false -> maps:put(schema_registry_auth, Auth, Map)
+  end,
+  
+  Map1 = maybe_read_auth_file(Map0),
+
+  maybe_closure_auth_password(Map1).
+
+maybe_string_to_atom(false) -> false;
+maybe_string_to_atom(Env) -> list_to_atom(Env).
+
+maybe_read_auth_file(#{schema_registry_auth := {Mechanism, File}} = Vars) ->
+  case read_file(File, 2) of
+    [Username, Password] -> 
+      maps:put(schema_registry_auth, {Mechanism, Username, Password}, Vars);
+    _Any -> 
+      error_logger:error_msg("Malformed authorization file ~s", [File]),
+      Vars
+  end;
+maybe_read_auth_file(Vars) -> Vars.
+
+read_file(File, NumberOfLines) -> 
+  {ok, Device} = file:open(File, [raw, read, read_ahead]),
```

Comment:
`file:read_file/1` can get all lines in one read, so there is no need to open and close.
otherwise read in a `try` clause and close in `after`

Reply:
Hi, @zmstone.

No problem, I've followed your suggestion, since it's simpler reading a file using the function you pointed. :+1: